src/d/actor/d_a_coming3.cpp

```diff
@@ -4,125 +4,471 @@
+    }
+    return false;
 }
 
 /* 00001150-000014DC       .text eff_break_tsubo__Q29daComing35Act_cFv */
 void daComing3::Act_c::eff_break_tsubo() {
-    /* Nonmatching */
+    static dBgS_ObjGndChk gnd;
+
+    cXyz sp40(
+        current.pos.x,
+        current.pos.y + 200.0f,
+        current.pos.z
+    );
+    gnd.SetPos(&sp40);
+    gnd.SetActorPid(fopAcM_GetID(this));
+    dComIfG_Bgsp()->GroundCross(&gnd);
+    u32 soundId = dComIfG_Bgsp()->GetMtrlSndId(gnd);
+    fopAcM_seStartCurrent(this, JA_SE_OBJ_COL_SWC_CHNS, soundId);
+
+    cXyz sp34(2.0f, 2.0f, 2.0f);
+    cXyz sp28(0.0f, 0.0f, 0.0f);
+    cXyz sp1C = current.pos;
+    cMtx_multVec(unk_410, &sp28, &sp1C);
+
+    J3DAnmTexPattern* texPattern;
+
+    J3DModelData* modelData = (J3DModelData*)dComIfG_getObjectRes("Always", ALWAYS_BDL_MPM_TUBO);
+    texPattern = (J3DAnmTexPattern*)dComIfG_getObjectRes("Always", ALWAYS_BTP_MPM_TUBO);
+    
+    JPABaseEmitter* baseEmitter = (JPABaseEmitter*)dComIfGp_particle_set(dPa_name::ID_COMMON_0017, &sp1C, NULL, &sp34);
+    if (baseEmitter != NULL) {
+        dPa_J3DmodelEmitter_c* modelEmitter;
+        modelEmitter = new dPa_J3DmodelEmitter_c(baseEmitter, modelData, tevStr, texPattern, 0, 0);
+        if (modelEmitter != NULL) {
+            dComIfGp_particle_addModelEmitter(modelEmitter);
+        }
+    }
+
+    GXColor sp18;
+    sp18.r = tevStr.mColorC0.r;
+    sp18.g = tevStr.mColorC0.g;
+    sp18.b = tevStr.mColorC0.b;
+    sp18.a = tevStr.mColorC0.a;
+    dPa_selectTexEcallBack* cb = &dPa_control_c::mTsubo[1];
+    dComIfGp_particle_set(dPa_name::ID_COMMON_0018, &sp1C, NULL, &sp34, 0xFF, cb, -1, &sp18, &tevStr.mColorK0);
 }
 
 /* 000014DC-00001590       .text coming_process_main__Q29daComing35Act_cFv */
 void daComing3::Act_c::coming_process_main() {
-    /* Nonmatching */
+    typedef void (daComing3::Act_c::*main_func)();
+    static main_func main_table[] = {
+        &daComing3::Act_c::coming_start_main,
+        &daComing3::Act_c::coming_game_main,
+        &daComing3::Act_c::coming_wait_main,
+    };
+
+    if (unk_444 >= 0 && unk_444 < 3) {
+        (this->*main_table[unk_444])();
+    }
 }
 
+static const f32 unused_4595[1] = {6000.0f};
+
 /* 00001590-0000167C       .text set_mtx__Q29daComing35Act_cFv */
 void daComing3::Act_c::set_mtx() {
-    /* Nonmatching */
+    cXyz sp14(0.0f, 390.0f, 0.0f);
+    cXyz sp08(2.0f, 2.0f, 2.0f);
+
+    unk_440->setBaseScale(sp08);
+    mDoMtx_stack_c::transS(current.pos);
+    mDoMtx_stack_c::ZXYrotM(shape_angle.x, shape_angle.y, shape_angle.z);
+    mDoMtx_stack_c::transM(sp14);
+    mDoMtx_stack_c::scaleM(sp08);
+    unk_440->setBaseTRMtx(mDoMtx_stack_c::get());
+    MTXCopy(mDoMtx_stack_c::get(), unk_410);
+    unk_440->calc();
 }
 
 /* 0000167C-000016F8       .text collision_init__Q29daComing35Act_cFv */
 void daComing3::Act_c::collision_init() {
-    /* Nonmatching */
+    mStts.Init(0xFF, 0xFF, this);
+    mCyl.Set(M_cyl_src);
+    mCyl.SetStts(&mStts);
+    mCyl.SetTgVec(cXyz::Zero);
```

Comment:
You can get this to compile by casting away the const instead.
```suggestion
    mCyl.SetTgVec((cXyz&)cXyz::Zero);
```